cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -197,29 +198,15 @@ func createVolumeAndDomain(machine *clusterv1.Machine, offset int, kubeClient ku
 }
 
 // deleteVolumeAndDomain deletes a domain and its referenced volume
-func deleteVolumeAndDomain(machine *clusterv1.Machine) error {
-	// decode config
-	machineProviderConfig, err := machineProviderConfigFromClusterAPIMachineSpec(&machine.Spec)
-	name := machine.Name
-	if err != nil {
-		return fmt.Errorf("error getting machineProviderConfig from spec: %v", err)
-	}
-
-	// build libvirtd client
-	client, err := libvirtutils.BuildClient(machineProviderConfig.URI)
-	if err != nil {
-		return fmt.Errorf("Failed to build libvirt client: %s", err)
-	}
-
-	// delete domain
-	if err := libvirtutils.DeleteDomain(name, client); err != nil {
+func deleteVolumeAndDomain(machine *clusterv1.Machine, client *libvirtutils.Client) error {
```

Comment:
This function does too much. From the callers perspective we should just `deleteDomain()` and it that fails it can go onto doing other things like `deleteVolume()`. And vice-versa. Trying to do both in one shot has the unexpected side effect of leaving one resource behind. If we're going to continue down this path we should fall- through and attempt to `deleteVolume()` if this call fails.

Reply:
I don't disagree at all, but seems out of scope here. I really just want to fix the connection leak. I only threw in the fix for the volume cleanup because it was trivial.

Reply:
> I only threw in the fix for the volume cleanup because it was trivial.

It was that that made me comment. :)

Reply:
Sure, let's keep changes discrete.